msal/src/main/java/com/microsoft/identity/client/PublicClientApplication.java

```diff
@@ -779,34 +675,54 @@ public void acquireToken(@NonNull final Activity activity,
                              @Nullable final String[] extraScopesToConsent,
                              @Nullable final String authority,
                              @NonNull final AuthenticationCallback callback) {
-        String loginHint = null;
 
-        if (null != account) {
-            loginHint = account.getUsername();
-        }
+        AcquireTokenParameters acquireTokenParameters = new AcquireTokenParameters();
```

Comment:
What do you think creating a private `acquireToken method` which accepts all the parameters and all the overloads just call this method and `AcquireTokenParameters` are created inside it to avoid code duplication like below

`    private void acquireToken(@NonNull final Activity activity,
                             @NonNull final String[] scopes,
                             @Nullable final IAccount account,
                             @Nullable final String loginHint,
                             @NonNull final UiBehavior uiBehavior,
                             @Nullable final List<Pair<String, String>> extraQueryParameters,
                             @Nullable final String[] extraScopesToConsent,
                             @Nullable final String authority,
                             @NonNull final AuthenticationCallback callback) {

        AcquireTokenParameters acquireTokenParameters = new AcquireTokenParameters();
        acquireTokenParameters.setLoginHint(loginHint);
        acquireTokenParameters.setAccount(account);
        acquireTokenParameters.setActivity(activity);
        acquireTokenParameters.setScopes(Arrays.asList(scopes));
        acquireTokenParameters.setUIBehavior(uiBehavior);
        acquireTokenParameters.setExtraQueryStringParameters(extraQueryParameters);
        acquireTokenParameters.setAccountRecord(getAccountRecord(acquireTokenParameters.getAccount()));
        acquireTokenParameters.setCallback(callback);

        acquireToken(acquireTokenParameters);


    }`

Reply:
@kreedula I'd suggest this trades code duplication for the method-with-too-many-parameters problem.

If we're looking to clean this up, I'd prefer a builder object. @shoatman can make the call, I'd say drop it on the backlog

Reply:
I didn't follow you comment.  The method with too many parameters problem could be resolved by removing all existing overloads save the acquireToken(acquireTokenParameters) one.  Is that what you're suggesting?  I was thinking that we don't want to take that scale of breaking change at this time, but I"m open to it.  Thoughts?

I believe the second part of the suggestion is to support the builder patter on the token parameters classes and i'm open to that as well.